core/transaction_watcher.md

```diff
@@ -0,0 +1,38 @@
+## TransactionWatcher
+
+TransactionWatcher allows one to continuously watch (monitor), by means of polling, the status of a given transaction.
+
+We need a 'TransactionFetcher` that fetches the transaction from the network.
+
+```
+interface TransactionFetcher:
+    get_transaction(tx_hash: string): TransactionOnNetwork;
+```
+
+```
+class TransactionWatcher:
```

Comment:
I got used to `TransactionWatcher` but if we consider that `Awaiter` is a better name we can change it. We should think about an `AccountAwaiter` and if we think it will be usefull we'll add it.

Reply:
Makes sense. People also got used to this term. All right, let's keep it as it is :+1: